src/rust/transpiler/lib.rs

```diff
@@ -0,0 +1,96 @@
+use swc_core::common::FileName;
+use swc_core::common::GLOBALS;
+use swc_core::common::Globals;
+use swc_core::common::Mark;
+use swc_core::common::SourceMap;
+use swc_core::common::comments::SingleThreadedComments;
+use swc_core::common::sync::Lrc;
+use swc_core::ecma::ast::Module;
+use swc_core::ecma::ast::Pass;
+use swc_core::ecma::ast::Program;
+use swc_core::ecma::codegen::to_code_default;
+use swc_core::ecma::parser::Parser;
+use swc_core::ecma::parser::StringInput;
+use swc_core::ecma::parser::Syntax;
+use swc_core::ecma::parser::TsSyntax;
+use swc_core::ecma::transforms::typescript::strip;
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("io error: {0:?}")]
+    Io(#[from] std::io::Error),
+
+    #[error("utf8 error: {0:?}")]
+    Utf8(#[from] std::string::FromUtf8Error),
+
+    #[error("parser error: {0:?}")]
+    Parser(swc_core::ecma::parser::error::Error),
+}
+
+// #[from] fails for this error for some reason
+impl From<swc_core::ecma::parser::error::Error> for Error {
+    fn from(err: swc_core::ecma::parser::error::Error) -> Self {
+        Self::Parser(err)
+    }
+}
+
+type Result<T> = std::result::Result<T, Error>;
+
+#[cxx::bridge(namespace = "workerd::rust::transpiler")]
+mod ffi {
+    extern "Rust" {
+        fn transpile_bytes(src: &[u8]) -> Result<String>;
+    }
+}
+
+fn transpile_bytes(src: &[u8]) -> Result<String> {
+    transpile_string(String::from_utf8_lossy(src).to_string())
+}
+
+fn transpile_string(src: String) -> Result<String> {
```

Comment:
Does this correctly preserve line numbers?
At least for when erasable syntax was used?